pkg/controller/job/job_controller.go

```diff
@@ -263,7 +261,12 @@ func (jm *Controller) addPod(obj interface{}) {
 		return
 	}
 
-	// Otherwise, it's an orphan. Get a list of all matching controllers and sync
+	// Otherwise, it's an orphan.
+	// Clean the finalizer.
+	if hasJobTrackingFinalizer(pod) {
+		jm.enqueueOrphanPod(pod)
```

Comment:
this is putting the pod into a different queue? will that remove the finalizer even if an ownerRef is added in the meantime by an adoption below?

Reply:
No, during the sync, we check whether the owner ref and job exists again.

Reply:
See the update for `syncOrphanPod` below